src/idl_gen_js.cpp

```diff
@@ -24,45 +25,73 @@
 namespace flatbuffers {
 
 static std::string GeneratedFileName(const std::string &path,
-                                     const std::string &file_name) {
-  return path + file_name + "_generated.js";
+                                     const std::string &file_name,
+                                     bool ts) {
+  return path + file_name + ((ts) ? ("_generated.ts") : ("_generated.js"));
 }
 
 namespace js {
 // Iterate through all definitions we haven't generate code for (enums, structs,
 // and tables) and output them to a single file.
 class JsGenerator : public BaseGenerator {
  public:
+  typedef std::unordered_set<std::string> ImportedFileSet;
+
   JsGenerator(const Parser &parser, const std::string &path,
-              const std::string &file_name)
-      : BaseGenerator(parser, path, file_name, "", "."){};
+              const std::string &file_name, bool ts)
+      : BaseGenerator(parser, path, file_name, "", ".")
+      , ts_(ts)
+  {
+  };
   // Iterate through all definitions we haven't generate code for (enums,
   // structs, and tables) and output them to a single file.
   bool generate() {
     if (IsEverythingGenerated()) return true;
 
-    std::string enum_code, struct_code, exports_code, code;
+    ImportedFileSet importedFiles;
+
+    std::string enum_code, struct_code, import_code, exports_code, code;
     generateEnums(&enum_code, &exports_code);
-    generateStructs(&struct_code, &exports_code);
+    generateStructs(&struct_code, &exports_code, importedFiles);
+    generateImportDependencies(&import_code, importedFiles);
 
     code = code + "// " + FlatBuffersGeneratedWarning();
 
     // Generate code for all the namespace declarations.
     GenNamespaces(&code, &exports_code);
 
     // Output the main declaration code from above.
+    if (ts_) {
+        code += import_code;
```

Comment:
Yes, it would be ignored, but then there's no point in generating it in the first place. As the old saying goes - some work is more work than no work :)